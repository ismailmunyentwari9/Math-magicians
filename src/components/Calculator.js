import './calc.css';
import Buttons from './buttons';

function calculator() {
  return (
    <div className="calc-container">

      <div className="output">
        <div className="prev-operand" />
        <div className="current-operand" />
      </div>
      <Buttons />

    </div>
  );
}
export default calculator;
