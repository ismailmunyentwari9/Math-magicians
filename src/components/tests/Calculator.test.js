import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';
import Calculator from '../Calculator';

describe('Calculator', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Calculator />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
