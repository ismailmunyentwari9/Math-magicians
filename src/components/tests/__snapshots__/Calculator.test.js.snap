// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  style={
    Object {
      "backgroundColor": "#E0E0E0",
      "padding": "20px",
    }
  }
>
  <div
    style={
      Object {
        "color": "white",
        "fontSize": "24px",
        "marginBottom": "20px",
      }
    }
  >
    Let us do some math!
  </div>
  <div
    style={
      Object {
        "backgroundColor": "white",
        "borderRadius": "10px",
        "padding": "20px",
      }
    }
  >
    <div
      style={
        Object {
          "marginBottom": "20px",
        }
      }
    >
      <div>
        <span>
          0
        </span>
      </div>
    </div>
    <div
      className="buttons"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        x
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        -
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        +
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;
